### PR TITLE
fix: contributions ui issues and search

### PR DESCRIPTION
**Pull request by kaelsky**
### Problem 

https://onlydust.slack.com/archives/C05Q707EKU3/p1695107150106749

### Solution 

- [x] Prevent displaying contributions <= 0
- [x] Substract codeReviews from total 
- [x] Fix contributions count on mobile view
- [x] Fix contributions count in contributors search 
- [x] Fix search filter in RewardFormSidePanel

### CODE REVIEW

File: frontend/src/pages/ProjectDetails/Contributors/ContributorsTable/Line.tsx [hunk 1]
```diff
@@ -23,6 +25,9 @@ export default function ContributorLine({
   onRewardGranted,
 }: Props) {
   const { T } = useIntl();
+  const isSm = useMediaQuery(`(min-width: ${viewportConfig.breakpoints.sm}px)`);
```
Comment (haydencleary) [inline frontend/src/pages/ProjectDetails/Contributors/ContributorsTable/Line.tsx:28]: Can't you use Tailwind for this media query.

File: frontend/src/pages/ProjectDetails/Rewards/RewardForm/WorkItemSidePanel/Issues/useFilteredWorkItems.ts [hunk 2]
```diff
@@ -15,8 +15,9 @@ export default function useFilteredContributions({ pattern = "", contributions }
 
   return filter(contributions, (contribution: ContributionFragment) => {
     if (!contribution.id) return;
-    // const contributionFullText = contribution.number.toString() + " " + contribution.title;
-    const contributionFullText = contribution.id;
+
+    const contributionType = contribution?.type === "ISSUE" ? contribution.githubIssue : contribution.githubPullRequest;
```
Comment (haydencleary) [inline frontend/src/pages/ProjectDetails/Rewards/RewardForm/WorkItemSidePanel/Issues/useFilteredWorkItems.ts:19]: Isn't there a constant or an enum for `"ISSUE"`?

File: frontend/src/pages/ProjectDetails/Contributors/ContributorsTable/Line.tsx (hunk 1 above)
Comment (kaelsky) [reply frontend/src/pages/ProjectDetails/Contributors/ContributorsTable/Line.tsx:28]: We could indeed. If we choose to do so we should refactor all occurrences of this hook in the project in another ticket

File: frontend/src/pages/ProjectDetails/Rewards/RewardForm/WorkItemSidePanel/Issues/useFilteredWorkItems.ts (hunk 2 above)
Comment (kaelsky) [reply frontend/src/pages/ProjectDetails/Rewards/RewardForm/WorkItemSidePanel/Issues/useFilteredWorkItems.ts]: https://github.com/onlydustxyz/marketplace/blob/87214b70122c67536e25043807ee0afdf02f5402/frontend/src/__generated/graphql.tsx#L9412-L9416

Yes `WorkItemType` !

**Review by PierreOucif — APPROVED** [review]

**Review by ofux — APPROVED** [review]

**Pull request merged**